dynamiq/connections/connections.py

```diff
@@ -855,6 +856,61 @@ def connect(self):
         pass
 
 
+class MilvusDeploymentType(str, enum.Enum):
+    """
+    Defines various deployment types for different Milvus deployments.
+
+    Attributes:
+        LOCAL_FILE (str): Represents a local file-based deployment.
+        DOCKER (str): Represents a Docker-based deployment on localhost.
+        ZILLIZ_CLOUD (str): Represents a cloud deployment on Zilliz Cloud.
+    """
+
+    LOCAL_FILE = "local_file"
+    DOCKER = "docker"
+    ZILLIZ_CLOUD = "zilliz_cloud"
+
+
+class Milvus(BaseConnection):
+    """
+    Represents a connection to the Milvus service.
+
+    Attributes:
+        type (Literal[ConnectionType.Milvus]): The type of connection, always 'Milvus'.
+        deployment_type (MilvusDeploymentType): The deployment type of the Milvus service
+        api_key (Optional[str]): The API key for Milvus on Zilliz Cloud, required only for cloud deployment.
+        uri (str): The URI for the Milvus instance (file path, Docker URL, or cloud URL).
+    """
+
+    type: Literal[ConnectionType.Milvus] = ConnectionType.Milvus
+    deployment_type: MilvusDeploymentType = MilvusDeploymentType.LOCAL_FILE
+    uri: str = Field(default_factory=partial(get_env_var, "MILVUS_URI", "http://localhost:19530"))
+    api_key: str | None = Field(default_factory=partial(get_env_var, "MILVUS_API_TOKEN", None))
+
+    def connect(self):
+        from pymilvus import MilvusClient
+
+        if self.deployment_type == MilvusDeploymentType.LOCAL_FILE:
```

Comment:
Let's move `FILE` validation to Pydantic model validator to do it on initialization step. Could be like:
```python
@model_validator(mode="after")
def validate_uri(self) -> "Milvus":
    if self.self.deployment_type == MilvusDeploymentType.LOCAL and not self.uri.endswith(".db"):
        raise ValueError("For local file deployment, URI should point to a local file ending with '.db'.")
    return self
 ```